File: Tutoriels/Java/Projets/code/GProject/src/manager/GPrint.java
//===============================================
package manager;
//===============================================
public class GPrint {
	//===============================================
	private static GPrint m_instance = null; 
	//===============================================
	public GPrint() { 

	} 	
	//===============================================
	public static GPrint Instance() { 
		if(m_instance == null) { 
			m_instance = new GPrint(); 
		}
		return m_instance; 
	} 
	//===============================================
	public void print(String data) { 
		System.out.print(data + "\n");
	} 
	//===============================================
}
//===============================================

File: php/class/GPrint.php
<?php   
    class GPrint {
        //===============================================
        private static $m_instance = null;
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            if(is_null(self::$m_instance)) {
                self::$m_instance = new GPrint();  
            }
            return self::$m_instance;
        }
        //===============================================
        public function printData($data) {
            echo $data;
        }
        //===============================================
    }
?>